composer/algorithms/swa/swa.py

```diff
@@ -86,17 +86,25 @@ def __init__(self, swa_start: float = 0.8, anneal_epochs: int = 10, swa_lr: Opti
         self.swa_scheduler = None
 
     def match(self, event: Event, state: State) -> bool:
-        """Run in Event.TRAINING_START, Event.TRAINING_END or if Event.EPOCH_END and epochs greater than or equal to `swa_start * max_epochs`
+        """Run in Event.TRAINING_START or if Event.EPOCH_END and steps greater than or equal to `swa_start * max_epochs * steps_per_epoch`
 
         Args:
             event (:class:`Event`): The current event.
             state (:class:`State`): The current state.
         Returns:
             bool: True if this algorithm should run now.
         """
-        should_start_swa = state.epoch >= int(self.hparams.swa_start * state.max_epochs)
-        return event in (Event.TRAINING_START, Event.TRAINING_END) or \
-             (event == Event.EPOCH_END and should_start_swa)
+
+        if state.max_epochs == 1 and event == Event.TRAINING_START:
+            should_start_swa = state.step >= int(self.hparams.swa_start * state.steps_per_epoch)
+            return (event == Event.TRAINING_START or should_start_swa)
+
+        else:
+            if event == Event.TRAINING_START:
+                should_start_swa = state.step >= int(self.hparams.swa_start * state.max_epochs * state.steps_per_epoch)
+
+            return (event == Event.TRAINING_START) or \
+                (event == Event.EPOCH_END and should_start_swa)
```

Comment:
```suggestion
                (event == Event.EPOCH_END)
```
if we're at the epoch end event, and swa_start is on [0, 1], then wouldn't it always be true by epoch end?